# 2.28
La $I$ være det bestemte integralet $I=\int_1^2 \frac{1}{x+x^2}dx$

a) Finn $I$ med trapesmetoden med fem delintervaller

In [6]:
n = 5
a = 1 
b = 2

dx = (b-a)/n

def f(x):
    return 1/(x+x**2)

summen = 0

for i in range(n):
    # høyre
    x_ir = a + (i+1) * dx

    # venstre
    x_il = a + i * dx

    # arelaet til trapes
    T_i = ((f(x_ir) + f(x_il))*dx)/(2)

    summen += T_i

print(summen)

0.2897075147075147


### b) Finn $I$ med en midtpunkttilnærming med fem delintervaller.

In [ ]:
# Imponerende kort kode, dette skal brukes på eksamen
f = lambda x: 1/x+x**2

midpoint_sum = lambda f, a, b, n: ((b-a)/n) * sum(f(a + (i + 1/2) * (b-a)/n) for i in range(n))

print(midpoint_sum(f, 1, 2, 5))

0.28667367742325617


### c) Hvilke av svarene i oppgave a og b ligger nærmest det du finneri CAS?

Det jeg får i CAS er 0.287

Da ser vi at trapesmetoden er nærmest